streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -72,8 +82,23 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
         } else {
             throw new InvalidStateStoreException("Cannot get state store " + storeName + " because the stream thread is " +
                                                      state + ", not RUNNING" +
-                                                     (includeStaleStores ? " or REBALANCING" : ""));
+                                                     (storeQueryParams.staleStoresEnabled() ? " or REBALANCING" : ""));
         }
     }
 
+    private TaskId createKeyTaskId(final String storeName, final Integer partition) {
+        if (partition == null) {
+            return null;
+        }
+        final List<String> sourceTopics = internalTopologyBuilder.stateStoreNameToSourceTopics().get(storeName);
+        final Set<String> sourceTopicsSet = sourceTopics.stream().collect(Collectors.toSet());
+        final Map<Integer, InternalTopologyBuilder.TopicsInfo> topicGroups = internalTopologyBuilder.topicGroups();
+        for (final Map.Entry<Integer, InternalTopologyBuilder.TopicsInfo> topicGroup : topicGroups.entrySet()) {
+            if (topicGroup.getValue().sourceTopics.containsAll(sourceTopicsSet)) {
+                return new TaskId(topicGroup.getKey(), partition.intValue());
+            }
+        }
+        throw new InvalidStateStoreException("Cannot get state store " + storeName + " because the requested partition " + partition + "is" +
+                                                "not available on this instance");
```

Comment:
nit: missing whitespace -> `isnot`

Reply:
\cc @vitojeng -- seems this is a new case we should cover in KIP-216?

Reply:
In IP-216, because we focus on **stream state** , not stream thread state, so I think this is the same thing like line 66-71, IMO. I would wrap these exception base on stream state at the upper calling level, e.g., CompositeReadOnlyXXXStore.

Does this make sense?

Reply:
For line 66-71 above, we will throw `StateStoreMigratedException`, right? So I guess, even if it's a slightly different case, I guess you a right that we can also throw a `StateStoreMigratedException` for this case, too.

Discussing this question, I am wondering if we should still cover the case when the user passes in a partition number that is not available? This would be a user error similar to passing in a wrong store name?

Reply:
oh...I just realized what you meant. Yes, we should cover this case in KIP-216, will update KIP.
Thanks @mjsax !